### DISCUSSION TITLE
Include click docs on devdocs.io

### DISCUSSION THREAD

**Discussion by Cimbali** [pallets/click#2677] [Q&A] [answered]
Hi!

I’m a regular user of both click and [devdocs.io](https://devdocs.io/) – I feel it would be great to have your documentation be integrated on there to make it more practically searchable offline.

Essentially, that service scrapes docs periodically, reformats them, and makes them available via a single interface.

Every page footer lists the original docs page, the license, and the copyright holders. For example the footer of the “[10 minutes to pandas](https://devdocs.io/pandas~1/user_guide/10min)” page is:

> © 2008–2022, AQR Capital Management, LLC, Lambda Foundry, Inc. and PyData Development Team
> Licensed under the 3-clause BSD License.
> https://pandas.pydata.org/pandas-docs/version/1.5.0/user_guide/10min.html 

It seems to me that the 3rd clause of the [BSD 3-Clause License](https://github.com/pallets/click/blob/main/LICENSE.rst), would require Pallets to provide explicit written consent for this integration to be possible. The clause is as follows:

> 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.

- To me a devdocs reproduction of your documentation could be considered a product derived from this software.
- The name of the copyright holder is used, appropriately, to attribute the work of making the source documentation.
- Whether that could count as “promoting” in some weird legalese sense of the word I’m not sure (I’m not a lawyer).

So before I start working on a PR to devdocs, would it make sense for whoever can take this kind of decision here (maybe @davidism ?) to reply including the click docs there is OK? Or am I overthinking this?

For what it’s worth, devdocs is free to use, ad-free, and hosted by a donor-supporter nonprofit, [freeCodeCamp](https://github.com/freeCodeCamp) (of which I am not part, I just use devdocs and occasionally contribute to it.)

**Comment by rafalkrupinski** [comment]
As a Click user, why would choose 3rd party service that is less responsive, potentially outdated and less adjusted for Click compared to the original website?

**Reply by Cimbali** [reply]
I use it mostly because it makes the docs available offline, more responsive, and searchable through the same interface as many other docs. I think of it more as a mirror with some extra features than a service though - it’s entirely free.

**Comment by Cimbali** [comment, marked as answer]
In the end I’ve included it with a specific mention that there is no endorsement, so we should be in the clear:

> © Copyright 2014 Pallets.
> Licensed under the BSD 3-Clause License.
> We are not supported nor endorsed by Pallets.
